src/plugins/discover/public/customizations/customization_types/flyout_customization.ts

```diff
@@ -6,6 +6,10 @@
  * Side Public License, v 1.
  */
 
+import { DataTableRecord } from '@kbn/discover-utils/types';
+import { DocViewFilterFn } from '@kbn/unified-doc-viewer/types';
+import React, { ComponentType } from 'react';
```

Comment:
```suggestion
import type { DataTableRecord } from '@kbn/discover-utils/types';
import type { DocViewFilterFn } from '@kbn/unified-doc-viewer/types';
import React, { type ComponentType } from 'react';
```
Nit

Reply:
Those types are exported as types and interfaces from their related module, so they shouldn't have runtime code.
As far as I know, the `import type` is useful to tell the compiler about runtime types (classes, enums, ...) that should only be used for type-checking to strip out the runtime code definition at build time, so it shouldn't matter here as per the [TS doc](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-8.html#type-only-imports-and-export).

I'll change them anyway as it doesn't hurt if this is the preference in the Discover plugin :)